src/Npgsql/TypeHandlers/DateTimeHandlers/DateHandler.cs

```diff
@@ -53,17 +67,35 @@ internal DateHandler(PostgresType postgresType, bool convertInfinityDateTime)
         /// <inheritdoc />
         public override DateTime Read(NpgsqlReadBuffer buf, int len, FieldDescription? fieldDescription = null)
         {
-            var npgsqlDate = ReadPsv(buf, len, fieldDescription);
-
-            if (npgsqlDate.IsFinite)
-                return (DateTime)npgsqlDate;
-            if (!_convertInfinityDateTime)
-                throw new InvalidCastException("Can't convert infinite date values to DateTime");
-            if (npgsqlDate.IsInfinity)
-                return DateTime.MaxValue;
-            return DateTime.MinValue;
+            const string infinityExceptionMessage = "Can't convert infinite date values to DateTime";
```

Comment:
These consts don't seem very interesting to me, easier to just inline the messages... But no strong feelings about it.